src/app/pages/account/user-details/user-details.component.ts

```diff
@@ -1,33 +1,106 @@
+import {User} from 'src/app/models/User';
+import {faPen} from '@fortawesome/free-solid-svg-icons';
+import {AbstractControl, FormControl, FormGroup, ValidationErrors, Validators} from "@angular/forms";
+import {UserService} from "../../../services/user-service.service";
+import {UserProfile} from "../../../models/user-profile";
 
 @Component({
   selector: 'app-user-details',
   templateUrl: './user-details.component.html',
   styleUrls: ['../account.component.css'],
 })
 export class UserDetailsComponent implements OnInit {
-  @Input() user: User = {
+  user: User/* = {
     email: '',
     firstName: '',
-    settings: { notifications: { text: false, email: false }, theme: 'light' },
-  };
+    settings: {notifications: {text: false, email: false}, theme: 'light'},
+  };*/
+
+  profileRegisterGroup: FormGroup;
+
   inputsDisabled: boolean = true;
   faPen = faPen;
-
   nameIcon = faIdCard;
   dateIcon = faCalendar;
   phoneIcon = faPhone;
   emailIcon = faEnvelope;
 
-  constructor() {}
+  constructor(private userService: UserService) {
+  }
+
+  ngOnInit(): void {
+    //TODO refactor to diff methods
+    this.userService.userDetails.subscribe((user: User) => {
+      this.user = user;
+      this.profileRegisterGroup = new FormGroup({
+        email: new FormControl(this.user.email, {
+          validators: [Validators.email, Validators.required],
+        }),
+        DOB: new FormControl(this.user.DOB, {
+          validators: [
+            Validators.required,
+            Validators.pattern(/^\d{4}-\d{2}-\d{2}$/),
+            (control: AbstractControl): ValidationErrors | null => {
+              let dateDiff = Date.now() - Date.parse(control.value);
+              let ageDate = new Date(dateDiff);
+              if (ageDate.getUTCFullYear() - 1970 >= 18) return null;
+              return {
+                tooYoung: {value: control.value},
+              };
+            },
+          ],
+        }),
+        firstName: new FormControl(this.user.firstName, {
+          validators: [Validators.minLength(3), Validators.required],
+        }),
+        lastName: new FormControl(this.user.lastName),
+        phone: new FormControl(this.user.phone, {
+          validators: [
+            (control: AbstractControl): ValidationErrors | null => {
+              if (
+                //TODO validate with regex
+                !control.value ||
+                control.value.length === 0 ||
+                (control.value.length === 10 && !isNaN(control.value))
+              ) {
+                return null;
+              }
+              return {invalidPhoneNumber: {value: control.value}};
+            },
+          ],
+        }),
+      });
+    });
+  }
+
+  profileUpdate() {
+    this.profileRegisterGroup.updateValueAndValidity();
+    if (this.profileRegisterGroup.valid) {
+      this.userService.updateProfile(this.updateInfo());
+    }
+
+  }
+
+  updateInfo(): UserProfile {
+    let updatedUser = new UserProfile();
```

Comment:
This is to pass only the profile info that need updating back. Mainly to avoid the password validation in the backend. We could maybe pass back a user DTO with a dummy password too since this doesn't do anything with it, or if there's a better way that I didn't think about